routes/api/project/firewall.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
-class CloverApi
-  hash_branch(:project_prefix, "firewall") do |r|
+class Clover
+  hash_branch(:api_project_prefix, "firewall") do |r|
```

Comment:
This  `api_` practice is perhaps the weirdest part of the patch for me. Is this a durable practice or transitory? I kinda understand what you were getting at in general, to streamline the AppMode conditional branches.

Reply:
This is transitory until the routing trees are fully merged.  Once the routing trees are fully merged, there will be no need for separate prefixes.

Reply:
can we see a spike of how this might work, perhaps choosing from one to three (but maybe staring with just one) of the more complicated convergences as a model? No need to make the commit history neat or do the tests, per prototyping. This way we can glimpse the future and put phasing in context.

Reply:
In the main route block, we currently have `r.hash_branches("api")` for the api routes, and `r.hash_branches("")` for the web routes.  This will combine and both will use `r.hash_branches("")` after all branches have finished merging.

In terms of how individual route files will work when merged, this PR has the project/location/vm and project/vm route blocks already merged.  The only change when this is finalized will be that there will single `hash_branch` call per file, instead of two calls (one for api and one for web).